assess2/AssessRecord.php

```diff
@@ -180,6 +183,17 @@ public function saveRecord() {
       }
       $stm = $this->DBH->prepare($query);
       $stm->execute($qarr);
+      if ($stm->rowCount()>0) {
+          $result = TeacherAuditLog::addTracking(
```

Comment:
checking for `$this->data['scoreoverride']==true` seems to work, and I also am recording score details

Reply:
With that, it looks like it's still going to record a log entry every time the assessment record is saved, if there is a scoreoverride set.  Also, it shouldn't be necessary to decode the scoreddata - it was already decoded, and it doesn't even look like you're using it.  The loadRecord call is also redundant - if we're saving the record, it's already loaded, and calling loadRecord again is causing it to next transactions and is triggering an error.

Reply:
TBH, I'm not 100% convinced this log entry is even useful, since the original grades are still present and visible when looking at the student's attempts, and it's obvious there's a grade override.  Logging provides no additional information.